cardano-constitution/src/Cardano/Constitution/Config.hs

```diff
@@ -0,0 +1,37 @@
+module Cardano.Constitution.Config
+    ( fromParamName
+    , asParamId
+    , fromParamId
+    , asParamName
+    , lookupPredValue
+    , module Export
+    ) where
+
+import Cardano.Constitution.Config.Default as Export
+import Cardano.Constitution.Config.Types as Export
+import Cardano.Constitution.Predicate
+import Cardano.Constitution.ProtocolParam
+
+import PlutusTx.AssocMap qualified as Tx
+import PlutusTx.Eq qualified as Tx
+import PlutusTx.List qualified as Tx
+
+{-# INLINABLE asParamId #-}
+{-# INLINABLE fromParamName #-}
+asParamId, fromParamName :: ParamName -> ConstitutionConfig -> Maybe ParamId
```

Comment:
Why give each function two names?

More importantly, this is an expensive operation and I don't know why it is needed. For each parameter, the json file should contain the param Id and valid range. The ScriptContext should also contain the param Id and the new value. Param name shouldn't ever be needed.

Reply:
Outdated.